### PR TITLE
Lint and format: Autofix + checks (pre-commit, pre-push and for those who want on CI)

### PR DESCRIPTION
**Pull request by maxime1992**
This PR adds the ability to stop worrying about code linting and formatting :smile:!

Focus on the code thanks to prettier :+1:.

When you're done with your work, before a commit just run `yarn run check:fix` and it'll solve all the fixable tslint failures + format the code with prettier.

There's also a commit hook that will check lint + fomat before you can commit.

If you wanna make sure no one bypass those checks on CI you might just run `yarn run check` for that.

(I'm using `yarn run ...` in my examples but will work with npm too as I've used `yarn-or-npm` module).

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,22 +7,46 @@
...
-    "@nestjs/common": "^4.0.0",
-    "@nestjs/core": "^4.0.0",
-    "@nestjs/microservices": "^4.0.0",
-    "@nestjs/testing": "^4.0.0",
-    "@nestjs/websockets": "^4.0.0",
-    "redis": "^2.7.1",
-    "reflect-metadata": "^0.1.10",
-    "rxjs": "^5.4.0",
-    "typescript": "^2.5.2"
+    "@nestjs/common": "4.0.0",
```
Comment (wbhob) [inline package.json]: why are you locking dependencies here? We're already on version 4.1.3.

Comment (maxime1992) [reply package.json]: No reason, I'll fix that tonight

**Comment by maxime1992** [comment]
@wbhob done :)

**Comment by maxime1992** [comment]
Could we merge this @wbhob? :)

**Comment by wbhob** [comment]
Looks good to me, I'm not a collaborator so I don't have merging privileges yet.

**Comment by maxime1992** [comment]
My bad! :sweat_smile: thanks for the review.

@kamilmysliwiec do you think we could merge this?

**Comment by kamilmysliwiec** [comment]
Hi @maxime1992,
Thanks for your work, but the idea behind this starter project is to provide as fewer things as possible to avoid needlessly confuses. I think more complex, extensive starters could be created independently from this repository, for example as a part of the CLI.

**Comment by wbhob** [comment]
@kamilmysliwiec I think this one should be used by the CLI. Additionally, we could do what the Ionic team is doing and have a repo of official starters (nestjs/starters) which contains examples.

**Pull request closed without merging**